src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -49,6 +59,106 @@ public async Task PutJavaScript()
 
             await PutInternal(validatedAsAdmin: false);
         }
+        
+        [RavenAction("/databases/*/admin/indexes/test", "POST", AuthorizationStatus.DatabaseAdmin, DisableOnCpuCreditsExhaustion = true)]
+        public async Task PutTestIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "Input");
+
+                if (input.TryGet("IndexDefinition", out BlittableJsonReaderObject indexDefinitionBlittable) == false)
```

Comment:
Instead of doing it this way (by manually reading the blittable) let's introduce a new class:

```
public class TestIndexParameters
{
    public IndexDefinition IndexDefinition { get; set; }

    public BlittableJsonReaderObject Query { get; set }
}
```

so then you'll be able to deserialize it using `JsonDeserializationServer.TestIndexParameters`

Reply:
Changed